velox/exec/AggregateWindow.cpp

```diff
@@ -341,7 +348,34 @@ class AggregateWindowFunction : public exec::WindowFunction {
     });
 
     // Set null values for empty (non valid) frames in the output block.
-    setNullEmptyFramesResults(validRows, resultOffset, result);
+    setEmptyFramesResult(validRows, resultOffset, result, emptyResult_);
+  }
+
+  // Precompute and save the aggregate output for empty input in emptyResult_.
+  // This value is returned for rows with empty frames.
+  void computeDefaultAggregateValue(
+      const std::string& name,
+      const TypePtr& resultType) {
+    constexpr int kRowSizeOffset = 8;
+    constexpr int kOffset = kRowSizeOffset + 8;
+    aggregate_->setOffsets(kOffset, 0, 1, kRowSizeOffset);
```

Comment:
Can you explain what these values are ?

Reply:
These offset values are obtained from the function `AggregationTestBase::testStreaming`. Having a different offset from the one used in the constructor `AggregateWindowFunction` ensures the group added in this function does not affect the aggregate result (`value` is calculated from `offset_`). I am not sure if a group once added to the aggregate can be removed (`aggregate_->clear()` does not achieve this), so I used different offsets instead.
Could you please share what a proper fix for this would be?